src/hooks/useMission.js

```diff
@@ -0,0 +1,51 @@
+import { useSelector } from "react-redux";
+
+const useMission = () => {
+  const currentMissionId = useSelector(
+    (state) => state.missions.selectedMission
+  );
+
+  const currentMission = () =>
+    currentMissionId
+      ? useSelector((state) => state.missions.list[missionIndex])
+      : null;

```

Comment:
Tu w sumie po prostu warto chyba trzymać currentMission jako wartość operacyjną w tym hooku:
```suggestion
  const currentMission = 
      useSelector((state) => state.missions.list.find( ({ id }) => id === state.missions.currentMissionId))
```
i warto tę funkcję wyciągnąć probably jako customowy selector.
... o ile to w ogóle w tej formie zadziała xD